docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -0,0 +1,366 @@
+
+Let's see what it has created for us.
+
+```yml title="data-structures/com.example/login.yaml"
+apiVersion: v1
+resourceType: data-structure
+meta:
+  hidden: false
+  schemaType: event
+  customData: {}
+data:
+  $schema: http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#
+  self:
+    vendor: com.example
+    name: login
+    format: jsonschema
+    version: 1-0-0
+  type: object
+  properties: {}
+  additionalProperties: false
+```
+* `apiVersion` should always be `v1`
+* `resourceType` should remain `data-structure`
+* `meta.hidden` directly relates to showing and hiding [in BDP Console UI](/docs/understanding-tracking-design/managing-your-data-structures/ui/#hiding-a-data-structure)
+* `meta.schemaType` can be `event` or `entity`
+* `meta.customData` is a map of strings to strings that can be used to send across any key/value pairs you'd like to associate with the data structure
+* `data` is the actual [snowplow self describing schema](/docs/pipeline-components-and-applications/iglu/common-architecture/self-describing-json-schemas/) that this data structure describes
+
+## Modify, validate and publish
+
+Firstly we'll add a property to our data structure definition. We'd like to know if a login succeeded or failed. Our modified `login.yaml` should look like this
+
+```yml {16,17} title="data-structures/com.example/login.yaml"
+apiVersion: v1
+resourceType: data-structure
+meta:
+  hidden: false
+  schemaType: event
+  customData: {}
+data:
+  $schema: http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#
+  self:
+    vendor: com.example
+    name: login
+    format: jsonschema
+    version: 1-0-0
+  type: object
+  properties:
+    result:
+      enum: [success, failure]
+  additionalProperties: false
+```
+
+### Validate
+
+We should validate our changes before we attempt to publish them. Let's do that
+
+```bash
+$ snowplow-cli ds validate data-structures/com.example/login.yaml
+```
+:::tip
+You can supply snowplow-cli with a directory and it will look for anything that looks like a data structure. Also given the default data structure directory is being used the previous command is equivalent to `snowplow-cli ds validate`.
+:::
+
+You should see output similar to this:
+```
+3:00PM INFO validating from paths=[data-structures/com.example/login.yaml]
+3:00PM INFO will create file=data-structures/com.example/login.yaml vendor=com.example name=login version=1-0-0
+3:00PM WARN validation file=data-structures/com.example/login.yaml
+  messages=
+  │ The schema is missing the "description" property (/properties/result)
+  │ The schema is missing the "description" property (/)
+```
+
+### Publish to development
+
+Apart from the missing descriptions everything looks good. We can fill them in later™. Let's go ahead and publish our data structure to our [development](/docs/managing-data-quality/testing-and-qa-workflows/) environment.
```

Comment:
I found that TM superscript funny but can we please remove it from the docs :)

Reply:
wrong tone, understood.